ls1intum/Artemis
# `Plagiarism checks`: Do not show the plagiarism case button if students have not yet been notified

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
Some students saw a button `Plagiarism Case` in the exercise details even they have not been notified yet.

### Description
This PR also improves the following:
* Validation of certain REST calls related to plagiarism checks and plagiarism cases
* Reduces the amount of unnecessary data sent to the client
* Improves the performance of some actions
* Improves the usability of the plagiarism case detail page (e.g. show the student notification immediately after posting it)
* Makes sure (client side) that instructors only determine the verdict after having notified the student

Thanks to @akesfeden for helping out in the implementation 👍 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Text Exercise

You basically need to check the whole plagiarism process (as described in the documentation)
1. Submit two larger identical texts with student1 and student2
2. Instructor: check for plagiarism (in course administration exercise details)
3. Student1 and student2: verify that the button `Plagiarism Case` is **not** shown in exercise details 
4. Instructor: notify the students by saving the post (you can keep the default text). 
5. Student1 and student2: verify that the button `Plagiarism Case` is now shown in exercise details
6. Student1 and student2: check that you can click on the button and see the anonymized comparison, then answer the post in both cases
7. Instructor: Set two different verdicts for the two cases: once plagiarism and once point deduction
8. Instructor: Check the course scores and make sure both decisions are reflected here accordingly (note: this was not affected in this PR, but should still work)
9. Student1 and student2: check that you can see the corresponding verdict


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
no actual changes

### Screenshots
no actual changes
